official/resnet/resnet.py

```diff
@@ -169,34 +183,38 @@ def building_block(inputs, filters, training, projection_shortcut, strides,
     The output tensor of the block.
   """
   shortcut = inputs
-  inputs = batch_norm_relu(inputs, training, data_format)
 
-  # The projection shortcut should come after the first batch norm and ReLU
-  # since it performs a 1x1 convolution.
   if projection_shortcut is not None:
     shortcut = projection_shortcut(inputs)
+    shortcut = batch_norm(inputs=shortcut, training=training,
+                          data_format=data_format)
 
   inputs = conv2d_fixed_padding(
       inputs=inputs, filters=filters, kernel_size=3, strides=strides,
       data_format=data_format)
-
   inputs = batch_norm_relu(inputs, training, data_format)
+
   inputs = conv2d_fixed_padding(
       inputs=inputs, filters=filters, kernel_size=3, strides=1,
       data_format=data_format)
+  inputs = batch_norm_relu(inputs, training, data_format,
+                           intermediate_addition=shortcut)
```

Comment:
You can replace this with:
```
inputs = batch_norm(inputs, training, data_format)
inputs = tf.nn.relu(inputs + shortcut)
```
after which `batch_norm_relu` doesn't need the `intermediate_addition` arg anymore (in fact, you can get rid of `batch_norm_relu` entirely if you like).